### PR TITLE
Small optimizations

### PR DESCRIPTION
**Pull request by robert-zaremba**
This is a cosmetic update.

+ Update documentation (I hope few updates here are making it more clear)
+ extract common functionality from 2 lengthy functions in `miner/worker.go` - this improves readability and can also help with testing.

### CODE REVIEW

File: consensus/clique/clique.go [hunk 1]
```diff
@@ -137,8 +137,9 @@ var (
 	errRecentlySigned = errors.New("recently signed")
 )
 
-// SignerFn is a signer callback function to request a header to be signed by a
-// backing account.
+// SignerFn hashes and signs the data to be signed by a backing account. It should point to the
+// accounts.Wallet.SignData method.
```
Comment (karalabe) [inline consensus/clique/clique.go:141]: This is an interface method, it should not point to anything explicit. It's up to the caller. Please delete that part.

File: consensus/clique/clique.go [hunk 2]
```diff
@@ -137,8 +137,9 @@ var (
...
+// The function arguments are: current address, mimeType and data to be hashed and signed.
```
Comment (karalabe) [inline consensus/clique/clique.go:142]: Please add parameter names to the method if you want to provide more information, it doesn't make sense to add this line as a comment vs. params.

File: consensus/consensus.go [hunk 3]
```diff
@@ -100,6 +100,8 @@ type Engine interface {
 	//
 	// Note, the method returns immediately and will send the result async. More
 	// than one result may also be returned depending on the consensus algorithm.
+	// Stop is triggered by closing the @stop channel hence, any receive from
+	// @stop will succeed without blocking.
```
Comment (karalabe) [inline consensus/consensus.go:104]: This newly added comment is not particularly helpful. The only operation you can do with this stop channel is to read from it, so there's not much reason to explicitly lay out that you should read from it and it means stop.

File: miner/worker.go [hunk 4]
```diff
@@ -288,6 +288,29 @@ func (w *worker) close() {
 	close(w.exitCh)
 }
 
+// recalcRecommit recalculates the resubmitting interval upon feedback.
+// @target is the dration vale (time.Duration as floating point) where we want to go.
```
Comment (karalabe) [inline miner/worker.go:292]: Please don't introduce a new variable notation. There' is no `@` notation in Go. I'm also not sure "where we want to go" provides more info that not adding anything.

File: miner/worker.go [hunk 5]
```diff
@@ -991,10 +985,30 @@ func (w *worker) commit(uncles []*types.Header, interval func(), update bool, st
...
+// deep copy current receipts. This is should be used whene an asynchronus process can modify
+// current context.
+func (w *worker) copyCurrentReceipts() []*types.Receipt {
+	receipts := make([]*types.Receipt, len(w.current.receipts))
+	for i, l := range w.current.receipts {
+		cpy := *l
+		receipts[i] = &cpy
+	}
+	return receipts
+}
```
Comment (karalabe) [inline miner/worker.go:997]: IMHO moving this method out doesn't help much. It was 5 lines in total, a trivial slice copy, that we don't even use in another place to warrant separation.

File: miner/worker.go [hunk 6]
```diff
@@ -991,10 +985,30 @@ func (w *worker) commit(uncles []*types.Header, interval func(), update bool, st
 	return nil
 }
 
+// deep copy current receipts. This is should be used whene an asynchronus process can modify
```
Comment (karalabe) [inline miner/worker.go:988]: The method doc does not conform to go style, please start with the method name, and expand from there.

File: miner/worker.go [hunk 7]
```diff
@@ -991,10 +985,30 @@ func (w *worker) commit(uncles []*types.Header, interval func(), update bool, st
...
 }
+
+// computes total consumed fees in ETH. Block transactions and receipts have to have the same order.
+func totalFees(block *types.Block, receipts []*types.Receipt) *big.Float {
+	feesWei := new(big.Int)
+	for i, tx := range block.Transactions() {
+		feesWei.Add(feesWei, new(big.Int).Mul(new(big.Int).SetUint64(receipts[i].GasUsed), tx.GasPrice()))
+	}
+	return new(big.Float).Quo(new(big.Float).SetInt(feesWei), new(big.Float).SetInt(big.NewInt(params.Ether)))
+}
```
Comment (karalabe) [inline miner/worker.go:1014]: This method is again a small utility code that's used solely for 1 log line. Moving it into a separate method makes things weirder because it makes the reader wonder what the importance is.

File: miner/worker.go (hunk 4 above)
Comment (robert-zaremba) [reply miner/worker.go:292]: OK -  make sense. 
I will removed the second line. Re `@` - sometimes I find it useful to quickly point to a parameter (consider: "@param" vs "parameter param".

File: miner/worker.go (hunk 5 above)
Comment (robert-zaremba) [reply miner/worker.go:997]: Look at  the usage:
```
receipts := w.copyCurrentReceipts()
```
vs 
```
	receipts := make([]*types.Receipt, len(w.current.receipts))
	for i, l := range w.current.receipts {
		cpy := *l
		receipts[i] = &cpy
	}
```


If you read functions, it's good to have this kind of abstraction. You don't need to focus on parsing the loop, and you quickly gasp that you just want to obtain a copy. 

Side comment: this file has lot of lengthy functions and in many places it's hard to follow it.

File: miner/worker.go (hunk 6 above)
Comment (robert-zaremba) [reply miner/worker.go]: OK. I didn't use the prefix because it's a private method.

File: miner/worker.go (hunk 7 above)
Comment (robert-zaremba) [reply miner/worker.go:1014]: My motivation is similar to the case with `copyCurrentReceipts`. This function is used in `commit` - the `for` loop is not necessary there (it is there only to create a log) and obfuscates the function, making slower to read. 

I general - I would be happy, and I hope others would also benefit, if we have more such abstractions.

**Comment by robert-zaremba** [comment]
Hi @karalabe , I've responded on all comments. Any chance we can move forward?

File: consensus/clique/clique.go (hunk 1 above)
Comment (robert-zaremba) [reply consensus/clique/clique.go:141]: It's highly coupled with `Wallet.SignData` behaviour, notably hashing the message before signing. The current `Clique` implementation depends on it:

+ `SignData` will hash the data under the hood (without your knowledge)
+ `ecrecovery` will not hash, and if you forget about it you will have a wrong result.

File: consensus/consensus.go (hunk 3 above)
Comment (robert-zaremba) [reply consensus/consensus.go:104]: Not really. I was doing a private updated, and I had to verify if the channel is closed or fed with information. There is a significant difference: closed channels will always respond with a zero value. None-closed channels will respond exactly once per data  inserted into the channel.

Comment (robert-zaremba) [reply consensus/consensus.go:104]: If the comment is not clear then I'm happy to edit it though.

File: consensus/clique/clique.go (hunk 2 above)
Comment (robert-zaremba) [reply consensus/clique/clique.go]: OK. Done.

File: consensus/clique/clique.go (hunk 1 above)
Comment (karalabe) [reply consensus/clique/clique.go]: This is an interface, whether the current mplementation is a specific code is not relevant.

Comment (robert-zaremba) [reply consensus/clique/clique.go]: I've updated the comment. I added a note about current implementation, but if you want I can remove it.

File: consensus/consensus.go (hunk 3 above)
Comment (karalabe) [reply consensus/consensus.go:104]: Please don't use `@` notation anywhere

Comment (robert-zaremba) [reply consensus/consensus.go:104]: I've removed it few days ago (the snippet above is market as outdated).

Comment (karalabe) [reply consensus/consensus.go:104]: Screenshot from 2020-05-20 10-36-32

Comment (robert-zaremba) [reply consensus/consensus.go]: Ah, apologise. I was looking at different file (`clique.go`). I've just updated the comment.

**Comment by robert-zaremba** [comment]
Hi, is anything still blocking this PR to be merged?

**Comment by robert-zaremba** [comment]
Hi, can we merge this PR?
